src/jailer/src/env.rs

```diff
@@ -674,11 +674,18 @@ impl Env {
 
         // Daemonize before exec, if so required (when the dev_null variable != None).
         if let Some(dev_null) = dev_null {
-            // Call setsid().
             // SAFETY: Safe because it's a library function.
-            SyscallReturnCode(unsafe { libc::setsid() })
-                .into_empty_result()
-                .map_err(JailerError::SetSid)?;
+            let pgid = unsafe { libc::getpgid(0) };
+            // SAFETY: Safe because it's a library function.
+            let pid = unsafe { libc::getpid() };
+
+            // Call setsid() only if jailer is not a process group leader.
+            if pgid != pid {
+                // SAFETY: Safe because it's a library function.
+                SyscallReturnCode(unsafe { libc::setsid() })
+                    .into_empty_result()
+                    .map_err(JailerError::SetSid)?;
+            }
```

Comment:
Why not do the normal daemonization process via double-fork like https://0xjet.github.io/3OHA/2022/04/11/post.html

I.e., by forking before, the `setsid()` will succeed.